### PR TITLE
Improve challenge deletion 

### PR DESCRIPTION
**Pull request by ziyadbagh**
#### **Overview**
This PR improves the challenge deletion process with UI updates, a confirmation dialog, and added tests.

#### **Key Changes**
1. **Delete Icon Color**:
   - Updated to red (`MaterialTheme.colorScheme.error`) for consistency with destructive actions.

2. **Confirmation Dialog**:
   - Added a dialog to confirm challenge deletion with "Yes" (delete) and "No" (cancel) options.
   - Styled buttons with theme colors for clarity.

3. **Strings Updates**:
   - Added new strings in `strings.xml` for dialog text and feedback.

4. **UI Tests**:
   - Added tests to verify the delete button triggers the dialog and dialog buttons function as expected.
  

### Notes for Reviewers:
The high percentage of code duplication is due to the confirmation dialog being the same as the one used in the friends list screen. This was done intentionally for consistency. Refactoring the dialog into a utility was deemed unnecessary, as it is only used in these two places.

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt [hunk 1]
```diff
@@ -294,18 +306,75 @@ fun OngoingChallengeCard(
...
         }
   }
+
+  // Confirmation dialog
+  ConfirmationDialog(showDialog = showDialog, onClickAction = onDeleteClick)
+}
+
+@SuppressLint("SuspiciousIndentation")
+@Composable
+fun ConfirmationDialog(showDialog: MutableState<Boolean>, onClickAction: () -> Unit) {
```
Comment (ThomasMPont) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:330]: ## Requested changes

This is too much to duplicate around, you should move this and the delete button into `Buttons.kt`, and create a new `@Composable` called `DialogButton()` or `DeleteButton()` or something.

**Review by ThomasMPont — CHANGES_REQUESTED** [review]
## Changes

You did exactly what your task asked for. Good job!

## Requested changes

I'm not too satisfied with the modularity and maintainability of your code. You'll find some changes I've requested in code below, that are marked with this section's title.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt [hunk 2]
```diff
@@ -294,18 +306,75 @@ fun OngoingChallengeCard(
...
                     }
               }
         }
   }
+
+  // Confirmation dialog
+  ConfirmationDialog(showDialog = showDialog, onClickAction = onDeleteClick)
+}
+
+@SuppressLint("SuspiciousIndentation")
```
[the remark this answers is not in the document]
Comment (ziyadbagh) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:328]: It's removed

**Comment by ThomasMPont** [comment]
@ziyadbagh 
I was expecting you to move the delete button into a `@Composable` in `Buttons.kt`, say `Buttons.DeleteButton()`. This button would contain it's own `Dialog()`. Then you would call that button in both the screens that use it.

If you still don't understand what I mean, we can relegate it to a future task. Even though it probably won't ever be done though.

Either way, could you quickly delete that unused import from SonarCloud at least, and I'll approve this PR.

**Comment by ziyadbagh** [comment]
> I was expecting you to move the delete button into a `@Composable` in `Buttons.kt`, say `Buttons.DeleteButton()`. This button would contain it's own `Dialog()`. Then you would call that button in both the screens that use it.
> 
> If you still don't understand what I mean, we can relegate it to a future task. Even though it probably won't ever be done though.
> 
> Either way, could you quickly delete that unused import from SonarCloud at least, and I'll approve this PR.

I've removed the unused import flagged by SonarCloud. Refactoring the delete button into Buttons.kt isn't necessary at this time since it's only used in two places, and this task is focused solely on improving the challenge deletion process.

**Review by ThomasMPont — APPROVED** [review]
> I've removed the unused import flagged by SonarCloud. Refactoring the delete button into Buttons.kt isn't necessary at this time since it's only used in two places, and this task is focused solely on improving the challenge deletion process.

Fair enough. Should be good to merge then.

**Pull request merged**